src/backlight/portfolio/portfolio.py

```diff
@@ -90,14 +90,99 @@ def construct_portfolio(
     return Portfolio(positions)
 
 
-def calculate_pl(pt: Portfolio, mkt: List[MarketData]) -> pd.DataFrame:
+# def calculate_pl(pt: Portfolio, mkt: List[MarketData]) -> pd.DataFrame:
+#     """
+#     Apply the sum on the homogenized portfolio
+#     args:
+#         - portfolio : a defined portfolio
+#         - mkt : list of marketdata for each asset
+#         - base_ccy : asset of reference
+
+#     """
+#     pl = pt.value()
+#     return pl.sum(axis=1)
+
+
+def homogenize_pl(
+    pt: Portfolio, mkt: List[MarketData], base_ccy: str = "USD"
+) -> Portfolio:
+    """
+    Normalize PL to one asset reference and sum
+    args:
+       - portfolio : a defined portfolio
+       - mkt : list of marketdata for each asset
+       - base_ccy : asset of reference
+                     for FX, one should pay attention to the family of asset that
+                     share the same base (for example EURJPY, USDJPY, GBPJPY)
+                     then an auto reference would be JPY in case of cross FX (EURUSD, USDJPY) we should convert USD pl from first asset
+                     to JPY using USDJPY market
+                      -> Job done a bit different : all assets are converted to base_ccy. The sum is still not done.    """
+    new_positions = []
+
+    # We compute the intersection of index between market datas and portfolio positions for later use
+    mkt_pos_intersection = mkt[0].index.intersection(pt._positions[0].index)
+    for position in pt._positions:
+        # ccy is the currency wich in which are expressed the element of the position, e.g. JPY for USDJPY
+        ccy = position.symbol[-3:]
```

Comment:
If we add the info `unit` in `position`, we do not need to parse strings.

Reply:
Although it is coding styles things, if we follow these things, the code become more readable automatically 😄 

- Avoid Deep Nesting e.g. at most, we can have 3 nests. c.f. https://code.tutsplus.com/tutorials/top-15-best-practices-for-writing-super-readable-code--net-8118
- Keep function length within less than 40 lines c.f. https://github.com/google/styleguide/blob/gh-pages/pyguide.md#318-function-length

Reply:
Ok thanks. I will apply that.